### PR TITLE
MOSIP-20457 Fixed sonar bugs.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/util/EncryptorUtil.java [hunk 1]
```diff
@@ -85,7 +85,9 @@ public String encrypt(byte[] data, String refId) {
                 ServiceError error = (ServiceError) responseDto.getErrors().get(0);
                 throw new PacketDecryptionFailureException(error.getMessage());
             }
-
+            if(responseDto != null || responseDto.getResponse() != null) {
+                DecryptResponseDto decryptResponseDto = mapper.convertValue(responseDto.getResponse(), DecryptResponseDto.class);
+            }
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/EncryptorUtil.java:91]: Please update the code as per the discussion.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/EncryptorUtil.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/EncryptorUtil.java:91]: Yes updated.

**Review by loganathan-sekaran — APPROVED** [review]

**Review by loganathan-sekaran — APPROVED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java [hunk 2]
```diff
@@ -299,12 +299,12 @@ public List<Map<String, String>> generateAudit(String rid) {
...
 		String langCode=null;
 		String mandatoryLanguages = env.getProperty("mosip.mandatory-languages");
-		if (!StringUtils.isBlank(mandatoryLanguages)) {
+		if (!StringUtils.isBlank(mandatoryLanguages) && !mandatoryLanguages.equalsIgnoreCase("null")) {
 			String[] lanaguages = mandatoryLanguages.split(",");
 			langCode = lanaguages[0];
 		} else {
 			String optionalLanguages = env.getProperty("mosip.optional-languages");
-			if (!StringUtils.isBlank(optionalLanguages)) {
+			if (!StringUtils.isBlank(optionalLanguages) && !optionalLanguages.equalsIgnoreCase("null")) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java]: Remove this check

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java [hunk 3]
```diff
@@ -245,7 +245,7 @@ private Set<String> getDataCapturedLanguages(JSONObject mapperIdentity, JSONObje
 
 	private List<String> getDefaultTemplateLanguages() {
 		String defaultLanguages = env.getProperty("mosip.default.template-languages");
-		if (!StringUtils.isBlank(defaultLanguages)) {
+		if (!StringUtils.isBlank(defaultLanguages) && !defaultLanguages.equalsIgnoreCase(null)) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java]: Remove this check

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java [hunk 4]
```diff
@@ -299,12 +299,12 @@ public List<Map<String, String>> generateAudit(String rid) {
 	public String getLanguageCode() {
 		String langCode=null;
 		String mandatoryLanguages = env.getProperty("mosip.mandatory-languages");
-		if (!StringUtils.isBlank(mandatoryLanguages)) {
+		if (!StringUtils.isBlank(mandatoryLanguages) && !mandatoryLanguages.equalsIgnoreCase("null")) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java]: Remove this check

File: resident/resident-service/src/main/java/io/mosip/resident/util/ResidentServiceRestClient.java [hunk 5]
```diff
@@ -317,8 +320,13 @@ private HttpEntity<Object> setRequestHeader(Object requestType, MediaType mediaT
 				HttpEntity<Object> httpEntity = (HttpEntity<Object>) requestType;
 				HttpHeaders httpHeader = httpEntity.getHeaders();
 				for (String key : httpHeader.keySet()) {
-					if (!(headers.containsKey("Content-Type") && Objects.equals(key, "Content-Type")))
-						headers.add(key, httpHeader.get(key).get(0));
+					if (!(headers.containsKey("Content-Type") && Objects.equals(key, "Content-Type"))){
+						if(httpHeader.isEmpty() || httpHeader.get(key) == null || httpHeader.get(key).isEmpty()
+						|| (httpHeader.get(key) != null && httpHeader.get(key).size() == 0) ||
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/ResidentServiceRestClient.java]: Correct the condition as '(httpHeader.get(key) != null && !httpHeader.get(key).isEmpty())'

File: resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java [hunk 6]
```diff
@@ -111,6 +111,9 @@ public String getIdSchema(Double version) throws ApisResourceAccessException, JS
 				idschema.put(version, responseString);
 		}
 
+		if(idschema.isEmpty() || idschema==null)
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java]: Add braces {} to if condition.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/Utilities.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/ResidentServiceRestClient.java (hunk 5 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/ResidentServiceRestClient.java]: Done

File: resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java (hunk 6 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java [hunk 7]
```diff
@@ -111,6 +111,9 @@ public String getIdSchema(Double version) throws ApisResourceAccessException, JS
 				idschema.put(version, responseString);
 		}
 
+		if(idschema.isEmpty() || idschema==null){
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java]: Null check should be before isEmpty()

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java (hunk 7 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/IdSchemaUtil.java]: yes corrected.

**Review by loganathan-sekaran — APPROVED** [review]

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**